File: src/lib-common/is-alnum.js
function is_alnum(x)
{
 if(!is_str(x))
  return false
  
 if(is_empty(x))
  return false

 for(const v of x)
 {
  if(is_alpha(v))
   continue
  
  if(is_digit(v))
   continue
  
  return false
 }
 
 return true
}

File: src/prj-core/lib-common/is/is-alnum.cs
fn is_alnum x
 if not is_str x
  ret false

 if is_empty x
  ret false

 for x
  //alpha, digit or underscore

  if same v "_"
  elseif is_alpha v
  elseif is_digit v
  else
   ret false
 end

 ret true
end